src/github3/repos/branch.py

```diff
@@ -121,15 +121,19 @@ def protect(self, enforcement=None, status_checks=None):
 
     @decorators.requires_auth
     def unprotect(self):
-        """Disable force push protection on this branch."""
-        edit = {'protection': {'enabled': False}}
-        resp = self._patch(self._api, json=edit,
-                           headers=self.PREVIEW_HEADERS)
-        json = self._json(resp, 200)
-        if self._boolean(resp, 200, 404):
-            self._update_attributes(json)
-            return True
-        return False
+        """Disable force push protection on this branch.
+
+        See: https://git.io/fxctW
+
+        :returns:
+            True if successful, False otherwise
+        :rtype:
+            bool
+        """
+        url = self._build_url('protection', base_url=self._api)
+        preview_key = 'required_approving_review_count'
+        headers = BranchProtection.PREVIEW_HEADERS_MAP[preview_key]
+        return self._boolean(self._delete(url, headers=headers), 204, 404)
```

Comment:
Mh, since the new API doesn't return anything back on `DELETE` we might have to clear the `original_protection` attribute ourselves.

Reply:
I don't have a feel for how that's handled in other places in the code, if we modify the local object based on things we think we've done, or if we just note that the user may need to refresh the object.

Reply:
I hate our modification of `self` in general so if we want to not modify it based on what we *think* has happened, that seems like movement in the right direction.

Eventually, I'd love to treat objects as immutable. I was initially worried about memory usage/etc but I think I was optimizing pre-emptively